### PR TITLE
Integrating shmem comms into camera pipeline

### PR DESCRIPTION
**Pull request by benkroop**

### CODE REVIEW

File: camera/camera_viewer_bq.cc [hunk 1]
```diff
@@ -16,8 +22,16 @@ void CameraViewerBq::loop() {
 
   // Wait until we have the latest image_message
   bool got_msg = false;
-  while (camera_subscriber_->read(image_message, 1)) {
-    got_msg = true;
+  if (camera_subscriber_)
+  {
```
Comment (jpanikulam) [inline camera/camera_viewer_bq.cc]: clang-format this

File: vision/fiducial_detection_balsaq.cc [hunk 2]
```diff
@@ -24,8 +30,16 @@ void FiducialDetectionBq::loop() {
 
   // Wait until we have the latest image_message
   bool got_msg = false;
-  while (subscriber_->read(image_message, 1)) {
-    got_msg = true;
+  if (subscriber_)
+  {
```
Comment (jpanikulam) [inline vision/fiducial_detection_balsaq.cc]: Clang-format this as well

File: camera/camera_viewer_bq.cc (hunk 1 above)
Comment (benkroop) [reply camera/camera_viewer_bq.cc]: :+1:

File: vision/fiducial_detection_balsaq.cc (hunk 2 above)
Comment (benkroop) [reply vision/fiducial_detection_balsaq.cc]: :+1:

**Pull request merged**